src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -78,52 +80,51 @@ public void showAllTaskEvents(){
 					itemToAdd = nextDeadlineTask;
 					deadlineTaskCounter++;
 				}
-		    }
-		    
-		    LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
-		    
-		    if(currentDate == null){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    else if(!currentDate.equals(tempDate)){
-		    	currentDate = tempDate;
-		    	addLabel(currentDate);
-		    }
-		    
-		    addTaskEvent(itemToAdd);
+			}
+
+			LocalDate tempDate = getLatestDate(currentDate, itemToAdd);
+
+			if(!currentDate.equals(tempDate)){
+				currentDate = tempDate;
+				addLabel(currentDate);
+			}
+
+			addedCounter++;
+			addTaskEvent(itemToAdd);
 		}
-		
+
 		for(FloatingTask task: masterList.getFloatingTasksList()){
+			addedCounter++;
 			addTaskEvent(task);
 		}
 		
+		return addedCounter;
 	}
-	
+
 	public int showTaskEventsFromDate(LocalDate selectedDate){
```

Comment:
You would still have to iterate through the list and compare events to find out which event starts on the date you want and then use that index :/